diffkemp/simpll/SMTBlockComparator.cpp

```diff
@@ -0,0 +1,842 @@
+    switch (s.check()) {
+    case z3::unsat:
+        return 0;
+    default:
+        // If SAT (blocks not equal), SMT solving may be run once again.
+        // Decrease the remaining time by the time taken.
+        auto elapsed = timeSinceEpochMillisec() - start;
+        if (config.SMTTimeout > 0) {
+            if (elapsed >= remainingTime) {
+                throw OutOfTimeException();
+            } else {
+                remainingTime -= elapsed;
+            }
+        }
+        return 1;
+    }
+}
+
+bool SMTBlockComparator::isInvertibleInst(BasicBlock::const_iterator Inst) {
+    if (auto *cmpInst = dyn_cast<CmpInst>(Inst))
+        for (const auto &user : cmpInst->users())
+            if (isa<BranchInst>(user))
+                return true;
+    return false;
+}
+
+bool SMTBlockComparator::hasPossiblyInverseCmp(BasicBlock::const_iterator Start,
+                                               BasicBlock::const_iterator End) {
+    while (Start != End) {
+        // A CMP result must be an output variable (outside the current snippet)
+        // and be used by a branch instruction in order for the inversion of
+        // the CMP instruction to make sense.
+        if (isOutputVar(Start, End) && isInvertibleInst(Start))
+            return true;
+        Start++;
+    }
+    return false;
+}
+
+void SMTBlockComparator::updateInverseCondList() {
+    for (const auto *LValue : inverted) {
+        auto serialNumber = fComp->sn_mapL[LValue];
+        if (fComp->mappedValuesBySn.find(serialNumber)
+            != fComp->mappedValuesBySn.end()) {
+            auto values = fComp->mappedValuesBySn[serialNumber];
+            fComp->inverseConditions.emplace(values);
+        }
+    }
+}
+
+int SMTBlockComparator::doCompare(BasicBlock::const_iterator &InstL,
+                                  BasicBlock::const_iterator &InstR) {
+    // Back-up the start of the snippet
+    auto StartL = InstL;
+    auto StartR = InstR;
+    auto BBL = const_cast<BasicBlock *>(InstL->getParent());
+    auto BBR = const_cast<BasicBlock *>(InstR->getParent());
+
+    // Instructions have been found to differ, undo the last comparison
+    fComp->undoLastInstCompare(InstL, InstR);
+
+    do {
+        // Update InstL and InstR to point to the end of the snippet.
+        // There may be multiple possible synchronization points in the
+        // functions and the first one may be the incorrect one. We need to
+        // check them all.
+        findSnippetEnd(InstL, InstR);
+
+        try {
+            int res = compareSnippets(StartL, InstL, StartR, InstR, false);
+
+            if (res == 0)
+                return res;
+
+            // Try to invert a cmp operation whose result is an output value.
+            // This aims to facilitate more complex refactorings than
+            // inverse-branch-condition pattern, e.g. replacing x < 101 with
+            // x > 101 (where x is an integer) and swapping branches.
```

Comment:
Shouldn't it be
```suggestion
            // inverse-branch-condition pattern, e.g. replacing x < 101 with
            // x > 100 (where x is an integer) and swapping branches.
```
?